[archiver] Error "block number mismatch" brings down the node

After running a node and prover-node for some time, the following error is hit:
```
aztec-node-1           | 2024-08-12T15:35:44.773Z aztec:sequencer [INFO] Submitted rollup block 18 with 0 transactions duration=1323.4801650000736ms (Submitter: 0x15d34aaf54267db7d7c367839aaf71a00a2c6a65)
aztec-node-1           | file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32
aztec-node-1           |             throw new Error('Block number mismatch. Expected: ' + expectedL2BlockNumber + ' but got: ' + blockNum + '.');
aztec-node-1           |                   ^
aztec-node-1           | 
aztec-node-1           | Error: Block number mismatch. Expected: 19 but got: 18.
aztec-node-1           |     at processL2BlockProcessedLogs (file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32:19)
aztec-node-1           |     at retrieveBlockMetadataFromRollup (file:///usr/src/yarn-project/archiver/dest/archiver/data_retrieval.js:25:40)
aztec-node-1           |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
aztec-node-1           |     at async Archiver.sync (file:///usr/src/yarn-project/archiver/dest/archiver/archiver.js:134:44)
aztec-node-1           |     at async poll (file:///usr/src/yarn-project/foundation/dest/promise/running-promise.js:22:17)
aztec-node-1           | 
aztec-node-1           | Node.js v18.19.1
aztec-prover-1         | file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32
aztec-prover-1         |             throw new Error('Block number mismatch. Expected: ' + expectedL2BlockNumber + ' but got: ' + blockNum + '.');
aztec-prover-1         |                   ^
aztec-prover-1         | 
aztec-prover-1         | Error: Block number mismatch. Expected: 19 but got: 18.
aztec-prover-1         |     at processL2BlockProcessedLogs (file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32:19)
aztec-prover-1         |     at retrieveBlockMetadataFromRollup (file:///usr/src/yarn-project/archiver/dest/archiver/data_retrieval.js:25:40)
aztec-prover-1         |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
aztec-prover-1         |     at async Archiver.sync (file:///usr/src/yarn-project/archiver/dest/archiver/archiver.js:134:44)
aztec-prover-1         |     at async poll (file:///usr/src/yarn-project/foundation/dest/promise/running-promise.js:22:17)
aztec-prover-1         | 
aztec-prover-1         | Node.js v18.19.1
```

This error is not handled gracefully and kills the node.

Larger trace:

```
aztec-node-1           | 2024-08-12T15:34:47.260Z aztec:prover:proving-orchestrator [INFO] Successfully proven block 14!
aztec-node-1           | 2024-08-12T15:34:47.859Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-prover-1         | 2024-08-12T15:34:47.927Z aztec:archiver [VERBOSE] Retrieved 1 new L2 blocks between L1 blocks 23 and 23.
aztec-prover-1         | 2024-08-12T15:34:47.934Z aztec:archiver [VERBOSE] Processed 1 new L2 blocks up to 14
aztec-prover-1         | 2024-08-12T15:34:47.936Z aztec:archiver [VERBOSE] Synced to L1 block 23
aztec-node-1           | 2024-08-12T15:34:48.314Z aztec:sequencer:publisher [INFO] Published L2 block to L1 rollup contract gasPrice=1263331993 gasUsed=828429 transactionHash=0x9f824b63ed23e6f488202a270db7b77b912413b8f3c55aaa65c5b8463bab97e4 calldataGas=32728 calldataSize=2692 txCount=1 blockNumber=14 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8 eventName=rollup-published-to-l1 blockHash=0x2688f51df5dd45668decc8cc515cb4c843624dd773219c8c20d000e57513641c
aztec-node-1           | 2024-08-12T15:34:48.315Z aztec:sequencer [INFO] Submitted rollup block 14 with 1 transactions duration=1347.294517999515ms (Submitter: 0x15d34aaf54267db7d7c367839aaf71a00a2c6a65)
aztec-prover-1         | 2024-08-12T15:34:48.972Z aztec:l2_block_downloader [VERBOSE] Received 1 blocks from archiver after querying from 13 limit 1 (proven false)
aztec-prover-1         | 2024-08-12T15:34:48.972Z aztec:l2_block_downloader [VERBOSE] Reached target block number 13
aztec-prover-1         | 2024-08-12T15:34:48.972Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-prover-1         | 2024-08-12T15:34:48.974Z aztec:merkle_trees [VERBOSE] Block 13 is not ours, rolling back world state and committing state from chain
aztec-prover-1         | 2024-08-12T15:34:49.082Z aztec:world_state [VERBOSE] Handled new L2 block eventName=l2-block-handled duration=110.50880299974233 isBlockOurs=false txCount=1 blockNumber=13 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8
aztec-prover-1         | 2024-08-12T15:34:49.082Z aztec:world_state [VERBOSE] Forking world state at 13
aztec-prover-1         | 2024-08-12T15:34:49.163Z aztec:block-proving-job [INFO] Starting block proving job fromBlock=14 toBlock=14
aztec-prover-1         | 2024-08-12T15:34:49.165Z aztec:block-proving-job [VERBOSE] Starting block processing number=14 blockHash=0x2688f51df5dd45668decc8cc515cb4c843624dd773219c8c20d000e57513641c lastArchive=0x033220de7d642003394a80827897b61b8d6194ca3e59a241014a7394cae9dd9e noteHashTreeRoot=0x12bdbeb43c9d77433c2f9a2ee614e0d91f1c25182bdedf3b1853b0245f58af9a nullifierTreeRoot=0x0dae74403cb6b23f21f8ba7df4bf0676a970c2e846a6afcdcd913781ab290a75 publicDataTreeRoot=0x1f6f44032c84e7b4bb2146af06e10083f0ab34cbaab13cdc0ee53e373cdb7a7e historicalHeader=0x2d10b0d9596d1512cafc5dbc68cf9c9de79cc0e284165a403f80c28e5dd22870 chainId=0x0000000000000000000000000000000000000000000000000000000000007a69 version=0x0000000000000000000000000000000000000000000000000000000000000001 blockNumber=0x000000000000000000000000000000000000000000000000000000000000000e slotNumber=0x0000000000000000000000000000000000000000000000000000000000000013 timestamp=0x0000000000000000000000000000000000000000000000000000000066ba2b32 coinbase=0x0000000000000000000000000000000000000000 feeRecipient=0x0000000000000000000000000000000000000000000000000000000000000000 gasFees=[object Object]
aztec-prover-1         | 2024-08-12T15:34:49.165Z aztec:prover:proving-orchestrator [INFO] Starting block 0x000000000000000000000000000000000000000000000000000000000000000e for slot 0x0000000000000000000000000000000000000000000000000000000000000013 with 2 transactions
aztec-prover-1         | 2024-08-12T15:34:49.182Z aztec:prover:proving-orchestrator [INFO] Received transaction: 0d28014d353cec4b31a37bb8675c0f755328965ec1e312e53e97139bca0695af
aztec-prover-1         | 2024-08-12T15:34:49.221Z aztec:block-proving-job [VERBOSE] Processed all txs for block blockNumber=14 blockHash=0x2688f51df5dd45668decc8cc515cb4c843624dd773219c8c20d000e57513641c
aztec-block-watcher-1  | Latest block: 14
aztec-block-watcher-1  | Proven block: 7
aztec-node-1           | 2024-08-12T15:34:56.822Z aztec:node [INFO] Simulating tx 0d23b4dfa0784080006a4a3fbec80c046b7b9c0a7c46d3e3722fec3782967146
aztec-bot-1            | 2024-08-12T15:34:56.858Z aztec:pxe_service [INFO] Executed local simulation for 0d23b4dfa0784080006a4a3fbec80c046b7b9c0a7c46d3e3722fec3782967146
aztec-node-1           | 2024-08-12T15:34:58.372Z aztec:node [INFO] Simulating tx 183b1c0c3c36e7c5eb59bf804ed93a76f9610651fb28f8ff156aa6085275d3da
aztec-bot-1            | 2024-08-12T15:34:58.407Z aztec:pxe_service [INFO] Executed local simulation for 183b1c0c3c36e7c5eb59bf804ed93a76f9610651fb28f8ff156aa6085275d3da
aztec-node-1           | 2024-08-12T15:34:59.878Z aztec:node [INFO] Simulating tx 0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e
aztec-bot-1            | 2024-08-12T15:35:00.012Z aztec:pxe_service [INFO] Sending transaction 0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e
aztec-node-1           | 2024-08-12T15:35:00.015Z aztec:node [INFO] Received tx 0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e
aztec-node-1           | 2024-08-12T15:35:00.015Z aztec:tx_pool [INFO] Adding tx with id 0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e eventName=tx-added-to-pool txHash=0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e noteEncryptedLogCount=2 encryptedLogCount=1 unencryptedLogCount=0 noteEncryptedLogSize=1040 encryptedLogSize=524 unencryptedLogSize=8 newCommitmentCount=2 newNullifierCount=2 proofSize=0 size=18041 feePaymentMethod=none classRegisteredCount=0
aztec-bot-1            | 2024-08-12T15:35:00.019Z aztec:bot [INFO] Transaction 0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e sent, not waiting for it to be mined
aztec-node-1           | 2024-08-12T15:35:00.573Z aztec:sequencer [INFO] Building block 15 with 1 transactions
aztec-node-1           | 2024-08-12T15:35:00.573Z aztec:prover:proving-orchestrator [INFO] Starting block 0x000000000000000000000000000000000000000000000000000000000000000f for slot 0x0000000000000000000000000000000000000000000000000000000000000014 with 2 transactions
aztec-node-1           | 2024-08-12T15:35:00.576Z aztec:prover:proving-orchestrator [INFO] Received transaction: 0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e
aztec-node-1           | 2024-08-12T15:35:01.925Z aztec:prover:proving-orchestrator [INFO] Successfully proven block 15!
aztec-node-1           | 2024-08-12T15:35:02.220Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-prover-1         | 2024-08-12T15:35:02.388Z aztec:archiver [VERBOSE] Retrieved 1 new L2 blocks between L1 blocks 24 and 24.
aztec-prover-1         | 2024-08-12T15:35:02.395Z aztec:archiver [VERBOSE] Processed 1 new L2 blocks up to 15
aztec-prover-1         | 2024-08-12T15:35:02.397Z aztec:archiver [VERBOSE] Synced to L1 block 24
aztec-node-1           | 2024-08-12T15:35:02.978Z aztec:sequencer:publisher [INFO] Published L2 block to L1 rollup contract gasPrice=1232233429 gasUsed=828465 transactionHash=0x5989fbb6acda92275c383b26bde2a36b065a8a3cf51a40991a7db3b492878321 calldataGas=32764 calldataSize=2692 txCount=1 blockNumber=15 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8 eventName=rollup-published-to-l1 blockHash=0x300c57790284f84874286f61b575b2d329ef71596d9de2b33a604e54d87d1971
aztec-node-1           | 2024-08-12T15:35:02.979Z aztec:sequencer [INFO] Submitted rollup block 15 with 1 transactions duration=1355.8803909998387ms (Submitter: 0x15d34aaf54267db7d7c367839aaf71a00a2c6a65)
aztec-prover-1         | 2024-08-12T15:35:04.170Z aztec:l2_block_downloader [VERBOSE] Received 1 blocks from archiver after querying from 14 limit 1 (proven false)
aztec-prover-1         | 2024-08-12T15:35:04.170Z aztec:l2_block_downloader [VERBOSE] Reached target block number 14
aztec-prover-1         | 2024-08-12T15:35:04.170Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-prover-1         | 2024-08-12T15:35:04.172Z aztec:merkle_trees [VERBOSE] Block 14 is not ours, rolling back world state and committing state from chain
aztec-prover-1         | 2024-08-12T15:35:04.284Z aztec:world_state [VERBOSE] Handled new L2 block eventName=l2-block-handled duration=113.87341600004584 isBlockOurs=false txCount=1 blockNumber=14 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8
aztec-prover-1         | 2024-08-12T15:35:04.284Z aztec:world_state [VERBOSE] Forking world state at 14
aztec-prover-1         | 2024-08-12T15:35:04.370Z aztec:block-proving-job [INFO] Starting block proving job fromBlock=15 toBlock=15
aztec-prover-1         | 2024-08-12T15:35:04.371Z aztec:block-proving-job [VERBOSE] Starting block processing number=15 blockHash=0x300c57790284f84874286f61b575b2d329ef71596d9de2b33a604e54d87d1971 lastArchive=0x152cce6a71fc7c50d39f1f66276d6a67f652e9865030952c19d18a8b91fbae54 noteHashTreeRoot=0x04afaa77f94da85cdeb49e8c6144c450556728feb94888bb9476ac7253d2906b nullifierTreeRoot=0x2f65b577c42fa36bbb0826b18b8f4f45a671fec90639081c45f7ba419680ab9f publicDataTreeRoot=0x1f6f44032c84e7b4bb2146af06e10083f0ab34cbaab13cdc0ee53e373cdb7a7e historicalHeader=0x2688f51df5dd45668decc8cc515cb4c843624dd773219c8c20d000e57513641c chainId=0x0000000000000000000000000000000000000000000000000000000000007a69 version=0x0000000000000000000000000000000000000000000000000000000000000001 blockNumber=0x000000000000000000000000000000000000000000000000000000000000000f slotNumber=0x0000000000000000000000000000000000000000000000000000000000000014 timestamp=0x0000000000000000000000000000000000000000000000000000000066ba2b3e coinbase=0x0000000000000000000000000000000000000000 feeRecipient=0x0000000000000000000000000000000000000000000000000000000000000000 gasFees=[object Object]
aztec-prover-1         | 2024-08-12T15:35:04.371Z aztec:prover:proving-orchestrator [INFO] Starting block 0x000000000000000000000000000000000000000000000000000000000000000f for slot 0x0000000000000000000000000000000000000000000000000000000000000014 with 2 transactions
aztec-prover-1         | 2024-08-12T15:35:04.390Z aztec:prover:proving-orchestrator [INFO] Received transaction: 0a48ae58904fe4265b789243a80cca7b679a85da3ebfc8e7598aac96e49be80e
aztec-prover-1         | 2024-08-12T15:35:04.442Z aztec:block-proving-job [VERBOSE] Processed all txs for block blockNumber=15 blockHash=0x300c57790284f84874286f61b575b2d329ef71596d9de2b33a604e54d87d1971
aztec-block-watcher-1  | Latest block: 15
aztec-block-watcher-1  | Proven block: 7
aztec-node-1           | 2024-08-12T15:35:11.644Z aztec:node [INFO] Simulating tx 159ea4f19537f6834f0e7abbf04fe3281c57d569ac32da0272e1884f53d7fc0c
aztec-bot-1            | 2024-08-12T15:35:11.891Z aztec:pxe_service [INFO] Executed local simulation for 159ea4f19537f6834f0e7abbf04fe3281c57d569ac32da0272e1884f53d7fc0c
aztec-node-1           | 2024-08-12T15:35:13.398Z aztec:node [INFO] Simulating tx 24d47daaa31ed5a5dd97777b89f51cce86b3f9a7ef297cba54c6059e8d84a347
aztec-bot-1            | 2024-08-12T15:35:13.849Z aztec:pxe_service [INFO] Executed local simulation for 24d47daaa31ed5a5dd97777b89f51cce86b3f9a7ef297cba54c6059e8d84a347
aztec-node-1           | 2024-08-12T15:35:15.334Z aztec:node [INFO] Simulating tx 25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39
aztec-bot-1            | 2024-08-12T15:35:15.657Z aztec:pxe_service [INFO] Sending transaction 25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39
aztec-node-1           | 2024-08-12T15:35:15.660Z aztec:node [INFO] Received tx 25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39
aztec-node-1           | 2024-08-12T15:35:15.661Z aztec:tx_pool [INFO] Adding tx with id 25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39 eventName=tx-added-to-pool txHash=25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39 noteEncryptedLogCount=2 encryptedLogCount=1 unencryptedLogCount=0 noteEncryptedLogSize=1040 encryptedLogSize=524 unencryptedLogSize=8 newCommitmentCount=2 newNullifierCount=2 proofSize=0 size=18041 feePaymentMethod=none classRegisteredCount=0
aztec-bot-1            | 2024-08-12T15:35:15.736Z aztec:bot [INFO] Transaction 25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39 sent, not waiting for it to be mined
aztec-node-1           | 2024-08-12T15:35:16.141Z aztec:sequencer [INFO] Building block 16 with 1 transactions
aztec-node-1           | 2024-08-12T15:35:16.141Z aztec:prover:proving-orchestrator [INFO] Starting block 0x0000000000000000000000000000000000000000000000000000000000000010 for slot 0x0000000000000000000000000000000000000000000000000000000000000015 with 2 transactions
aztec-node-1           | 2024-08-12T15:35:16.145Z aztec:prover:proving-orchestrator [INFO] Received transaction: 25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39
aztec-block-watcher-1  | Latest block: 15
aztec-block-watcher-1  | Proven block: 7
aztec-node-1           | 2024-08-12T15:35:17.489Z aztec:prover:proving-orchestrator [INFO] Successfully proven block 16!
aztec-prover-1         | 2024-08-12T15:35:18.192Z aztec:archiver [VERBOSE] Retrieved 1 new L2 blocks between L1 blocks 25 and 25.
aztec-prover-1         | 2024-08-12T15:35:18.292Z aztec:archiver [VERBOSE] Processed 1 new L2 blocks up to 16
aztec-prover-1         | 2024-08-12T15:35:18.295Z aztec:archiver [VERBOSE] Synced to L1 block 25
aztec-node-1           | 2024-08-12T15:35:18.385Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-node-1           | 2024-08-12T15:35:18.540Z aztec:sequencer:publisher [INFO] Published L2 block to L1 rollup contract gasPrice=1204807561 gasUsed=828441 transactionHash=0x24f3138960efcb6aae0b0ff38b47dff07b1ae21627c0b08b97697c89318f4269 calldataGas=32740 calldataSize=2692 txCount=1 blockNumber=16 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8 eventName=rollup-published-to-l1 blockHash=0x0e068d8d5c0a556fa8dcd769c203ef211c7ed8ccbababbd1b03d22c78049ad43
aztec-node-1           | 2024-08-12T15:35:18.540Z aztec:sequencer [INFO] Submitted rollup block 16 with 1 transactions duration=1350.8911790000275ms (Submitter: 0x15d34aaf54267db7d7c367839aaf71a00a2c6a65)
aztec-prover-1         | 2024-08-12T15:35:19.375Z aztec:l2_block_downloader [VERBOSE] Received 1 blocks from archiver after querying from 15 limit 1 (proven false)
aztec-prover-1         | 2024-08-12T15:35:19.375Z aztec:l2_block_downloader [VERBOSE] Reached target block number 15
aztec-prover-1         | 2024-08-12T15:35:19.375Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-prover-1         | 2024-08-12T15:35:19.377Z aztec:merkle_trees [VERBOSE] Block 15 is not ours, rolling back world state and committing state from chain
aztec-prover-1         | 2024-08-12T15:35:20.927Z aztec:world_state [VERBOSE] Handled new L2 block eventName=l2-block-handled duration=1551.9581779995933 isBlockOurs=false txCount=1 blockNumber=15 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8
aztec-prover-1         | 2024-08-12T15:35:20.927Z aztec:world_state [VERBOSE] Forking world state at 15
aztec-prover-1         | 2024-08-12T15:35:21.347Z aztec:block-proving-job [INFO] Starting block proving job fromBlock=16 toBlock=16
aztec-prover-1         | 2024-08-12T15:35:21.348Z aztec:block-proving-job [VERBOSE] Starting block processing number=16 blockHash=0x0e068d8d5c0a556fa8dcd769c203ef211c7ed8ccbababbd1b03d22c78049ad43 lastArchive=0x300b9dba707e4e35c6549eb593245165e3abe4ceb54b09c4e1b03b04c5cd0a6a noteHashTreeRoot=0x1c42b0b852a6dd162e90cd04d980dc27fff62c69eb6027dd3026728aa9a03d75 nullifierTreeRoot=0x2bf2eaa4ee24cd6baf8477008ee08aecc0077ef8e65b516644a25fac33623e8e publicDataTreeRoot=0x1f6f44032c84e7b4bb2146af06e10083f0ab34cbaab13cdc0ee53e373cdb7a7e historicalHeader=0x300c57790284f84874286f61b575b2d329ef71596d9de2b33a604e54d87d1971 chainId=0x0000000000000000000000000000000000000000000000000000000000007a69 version=0x0000000000000000000000000000000000000000000000000000000000000001 blockNumber=0x0000000000000000000000000000000000000000000000000000000000000010 slotNumber=0x0000000000000000000000000000000000000000000000000000000000000015 timestamp=0x0000000000000000000000000000000000000000000000000000000066ba2b4a coinbase=0x0000000000000000000000000000000000000000 feeRecipient=0x0000000000000000000000000000000000000000000000000000000000000000 gasFees=[object Object]
aztec-prover-1         | 2024-08-12T15:35:21.349Z aztec:prover:proving-orchestrator [INFO] Starting block 0x0000000000000000000000000000000000000000000000000000000000000010 for slot 0x0000000000000000000000000000000000000000000000000000000000000015 with 2 transactions
aztec-prover-1         | 2024-08-12T15:35:21.391Z aztec:prover:proving-orchestrator [INFO] Received transaction: 25820354af9482cce1cc69a877f5ca500c348ae4cae8c8db5b5e5ff7115cff39
aztec-prover-1         | 2024-08-12T15:35:21.449Z aztec:block-proving-job [VERBOSE] Processed all txs for block blockNumber=16 blockHash=0x0e068d8d5c0a556fa8dcd769c203ef211c7ed8ccbababbd1b03d22c78049ad43
aztec-node-1           | 2024-08-12T15:35:27.259Z aztec:node [INFO] Simulating tx 2c78b14922df30ad9975237ef9011d65d8597be622d93a24b9cde207985f0997
aztec-bot-1            | 2024-08-12T15:35:27.297Z aztec:pxe_service [INFO] Executed local simulation for 2c78b14922df30ad9975237ef9011d65d8597be622d93a24b9cde207985f0997
aztec-node-1           | 2024-08-12T15:35:28.816Z aztec:node [INFO] Simulating tx 066890c5cfa255528fb1e1e862e4a6f21329aa4765e0172c36ba1e4fbbb42fc8
aztec-bot-1            | 2024-08-12T15:35:28.855Z aztec:pxe_service [INFO] Executed local simulation for 066890c5cfa255528fb1e1e862e4a6f21329aa4765e0172c36ba1e4fbbb42fc8
aztec-block-watcher-1  | Latest block: 16
aztec-block-watcher-1  | Proven block: 7
aztec-node-1           | 2024-08-12T15:35:30.350Z aztec:node [INFO] Simulating tx 02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7
aztec-bot-1            | 2024-08-12T15:35:30.389Z aztec:pxe_service [INFO] Sending transaction 02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7
aztec-node-1           | 2024-08-12T15:35:30.392Z aztec:node [INFO] Received tx 02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7
aztec-node-1           | 2024-08-12T15:35:30.392Z aztec:tx_pool [INFO] Adding tx with id 02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7 eventName=tx-added-to-pool txHash=02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7 noteEncryptedLogCount=2 encryptedLogCount=1 unencryptedLogCount=0 noteEncryptedLogSize=1040 encryptedLogSize=524 unencryptedLogSize=8 newCommitmentCount=2 newNullifierCount=2 proofSize=0 size=18041 feePaymentMethod=none classRegisteredCount=0
aztec-bot-1            | 2024-08-12T15:35:30.397Z aztec:bot [INFO] Transaction 02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7 sent, not waiting for it to be mined
aztec-node-1           | 2024-08-12T15:35:30.835Z aztec:sequencer [INFO] Building block 17 with 1 transactions
aztec-node-1           | 2024-08-12T15:35:30.835Z aztec:prover:proving-orchestrator [INFO] Starting block 0x0000000000000000000000000000000000000000000000000000000000000011 for slot 0x0000000000000000000000000000000000000000000000000000000000000016 with 2 transactions
aztec-node-1           | 2024-08-12T15:35:30.839Z aztec:prover:proving-orchestrator [INFO] Received transaction: 02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7
aztec-prover-agent-1   | 2024-08-12T15:35:31.050Z aztec:bb-prover [VERBOSE] TubeCircuit (prove) BB out - downloading grumpkin crs...
aztec-node-1           | 2024-08-12T15:35:32.192Z aztec:prover:proving-orchestrator [INFO] Successfully proven block 17!
aztec-node-1           | 2024-08-12T15:35:33.294Z aztec:sequencer:publisher [INFO] Published L2 block to L1 rollup contract gasPrice=1180620541 gasUsed=828453 transactionHash=0xa6e0c5389a5c289a8599163f20c93e730bee947c9d506db175cff760bc680a45 calldataGas=32752 calldataSize=2692 txCount=1 blockNumber=17 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8 eventName=rollup-published-to-l1 blockHash=0x19e80147f4629ce39163380deb15e8d08df72475a5e22b5f8bbb67853e744762
aztec-node-1           | 2024-08-12T15:35:33.294Z aztec:sequencer [INFO] Submitted rollup block 17 with 1 transactions duration=1359.36206900049ms (Submitter: 0x15d34aaf54267db7d7c367839aaf71a00a2c6a65)
aztec-prover-1         | 2024-08-12T15:35:33.306Z aztec:archiver [VERBOSE] Retrieved 1 new L2 blocks between L1 blocks 26 and 26.
aztec-prover-1         | 2024-08-12T15:35:33.313Z aztec:archiver [VERBOSE] Processed 1 new L2 blocks up to 17
aztec-prover-1         | 2024-08-12T15:35:33.314Z aztec:archiver [VERBOSE] Synced to L1 block 26
aztec-node-1           | 2024-08-12T15:35:33.391Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-prover-agent-1   | 2024-08-12T15:35:34.137Z aztec:bb-prover [VERBOSE] TubeCircuit (prove) BB out - File is empty or there's an error reading it: /usr/src/yarn-project/bb/tmp-6JZvPl/client_ivc_proof
aztec-prover-agent-1   | 2024-08-12T15:35:34.257Z aztec:bb-prover [ERROR] Failed to generate proof for tube proof: Failed to generate proof. Exit code 1. Signal null.
aztec-prover-agent-1   | 2024-08-12T15:35:34.259Z aztec:prover-client:prover-agent [ERROR] Error processing proving job id=7c689027 type=TUBE_PROOF: Error: Failed to generate proof. Exit code 1. Signal null.
aztec-prover-agent-1   |     at BBNativeRollupProver.generateTubeProofWithBB (file:///usr/src/yarn-project/bb-prover/dest/prover/bb_prover.js:276:27)
aztec-prover-agent-1   |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
aztec-prover-agent-1   |     at async operation (file:///usr/src/yarn-project/bb-prover/dest/prover/bb_prover.js:317:43)
aztec-prover-agent-1   |     at async runInDirectory (file:///usr/src/yarn-project/foundation/dest/fs/run_in_dir.js:10:16)
aztec-prover-agent-1   |     at async BBNativeRollupProver.getTubeProof (file:///usr/src/yarn-project/bb-prover/dest/prover/bb_prover.js:331:24)
aztec-prover-agent-1   |     at async elapsed (file:///usr/src/yarn-project/foundation/dest/timer/elapsed.js:9:20)
aztec-prover-agent-1   |     at async ProverAgent.work (file:///usr/src/yarn-project/prover-client/dest/prover-agent/prover-agent.js:94:36): Error Failed to generate proof. Exit code 1. Signal null.
aztec-prover-1         | 2024-08-12T15:35:34.261Z aztec:prover-client:prover-pool:queue [WARN] Job id=7c689027 type=TUBE_PROOF failed with error: Failed to generate proof. Exit code 1. Signal null.. Retry 2/3
aztec-prover-agent-1   | 2024-08-12T15:35:34.315Z aztec:bb-prover [INFO] Generated witness circuitName=empty-nested duration=9.291203000582755 inputSize=0 outputSize=0 eventName=circuit-witness-generation
aztec-prover-1         | 2024-08-12T15:35:34.353Z aztec:l2_block_downloader [VERBOSE] Received 1 blocks from archiver after querying from 16 limit 1 (proven false)
aztec-prover-1         | 2024-08-12T15:35:34.353Z aztec:l2_block_downloader [VERBOSE] Reached target block number 16
aztec-prover-1         | 2024-08-12T15:35:34.353Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-prover-1         | 2024-08-12T15:35:34.355Z aztec:merkle_trees [VERBOSE] Block 16 is not ours, rolling back world state and committing state from chain
aztec-prover-1         | 2024-08-12T15:35:34.463Z aztec:world_state [VERBOSE] Handled new L2 block eventName=l2-block-handled duration=109.97028900031 isBlockOurs=false txCount=1 blockNumber=16 noteEncryptedLogLength=1040 noteEncryptedLogCount=2 encryptedLogLength=524 encryptedLogCount=1 unencryptedLogCount=0 unencryptedLogSize=8
aztec-prover-1         | 2024-08-12T15:35:34.463Z aztec:world_state [VERBOSE] Forking world state at 16
aztec-prover-1         | 2024-08-12T15:35:34.557Z aztec:block-proving-job [INFO] Starting block proving job fromBlock=17 toBlock=17
aztec-prover-1         | 2024-08-12T15:35:34.558Z aztec:block-proving-job [VERBOSE] Starting block processing number=17 blockHash=0x19e80147f4629ce39163380deb15e8d08df72475a5e22b5f8bbb67853e744762 lastArchive=0x0c9b2c34cdce462b256b915456f18bb0198c1f5dcb9c354f0bd2e37765939da2 noteHashTreeRoot=0x0619e7b700300ae2b0c08fb86e85d95a116692235f8c970cc35e5ae4aa9b5358 nullifierTreeRoot=0x0a3811c49eebd6686a40a5d95e681b6eb865fa014d7ee9d7213d15d7bee43165 publicDataTreeRoot=0x1f6f44032c84e7b4bb2146af06e10083f0ab34cbaab13cdc0ee53e373cdb7a7e historicalHeader=0x0e068d8d5c0a556fa8dcd769c203ef211c7ed8ccbababbd1b03d22c78049ad43 chainId=0x0000000000000000000000000000000000000000000000000000000000007a69 version=0x0000000000000000000000000000000000000000000000000000000000000001 blockNumber=0x0000000000000000000000000000000000000000000000000000000000000011 slotNumber=0x0000000000000000000000000000000000000000000000000000000000000016 timestamp=0x0000000000000000000000000000000000000000000000000000000066ba2b56 coinbase=0x0000000000000000000000000000000000000000 feeRecipient=0x0000000000000000000000000000000000000000000000000000000000000000 gasFees=[object Object]
aztec-prover-1         | 2024-08-12T15:35:34.559Z aztec:prover:proving-orchestrator [INFO] Starting block 0x0000000000000000000000000000000000000000000000000000000000000011 for slot 0x0000000000000000000000000000000000000000000000000000000000000016 with 2 transactions
aztec-prover-1         | 2024-08-12T15:35:34.576Z aztec:prover:proving-orchestrator [INFO] Received transaction: 02e6061e3a7f196c54924dac70143f8741be419db95c1c1eab78a192676e3cf7
aztec-prover-1         | 2024-08-12T15:35:34.623Z aztec:block-proving-job [VERBOSE] Processed all txs for block blockNumber=17 blockHash=0x19e80147f4629ce39163380deb15e8d08df72475a5e22b5f8bbb67853e744762
aztec-prover-agent-1   | 2024-08-12T15:35:35.634Z aztec:bb-prover [INFO] Generated proof for EmptyNestedArtifact in 1318 ms, size: 409 fields circuitName=empty-nested circuitSize=32 duration=1317.3195989998057 inputSize=0 proofSize=13092 eventName=circuit-proving numPublicInputs=16
aztec-prover-agent-1   | 2024-08-12T15:35:35.635Z aztec:bb-prover [VERBOSE] BB out - Executing BB with: verify_ultra_honk -p /usr/src/yarn-project/bb/tmp-Htw2Xy/proof -k /usr/src/yarn-project/bb/tmp-Htw2Xy/vk
aztec-prover-agent-1   | 2024-08-12T15:35:35.672Z aztec:bb-prover [INFO] Successfully verified proof from key in 36.645440000109375 ms
aztec-prover-agent-1   | 2024-08-12T15:35:35.719Z aztec:bb-prover [INFO] Generated witness circuitName=private-kernel-empty duration=41.82064600009471 inputSize=30777 outputSize=16414 eventName=circuit-witness-generation
aztec-block-watcher-1  | Latest block: 17
aztec-block-watcher-1  | Proven block: 7
aztec-node-1           | 2024-08-12T15:35:42.419Z aztec:sequencer [INFO] Building block 18 with 0 transactions
aztec-node-1           | 2024-08-12T15:35:42.420Z aztec:prover:proving-orchestrator [INFO] Starting block 0x0000000000000000000000000000000000000000000000000000000000000012 for slot 0x0000000000000000000000000000000000000000000000000000000000000017 with 2 transactions
aztec-node-1           | 2024-08-12T15:35:42.506Z aztec:node [INFO] Simulating tx 1e95724451fbde4a0be28d72daacd0d94e88fa19f97c9c83a7db5ff911253f69
aztec-bot-1            | 2024-08-12T15:35:42.556Z aztec:pxe_service [INFO] Executed local simulation for 1e95724451fbde4a0be28d72daacd0d94e88fa19f97c9c83a7db5ff911253f69
aztec-prover-agent-1   | 2024-08-12T15:35:43.560Z aztec:bb-prover [INFO] Generated proof for PrivateKernelEmptyArtifact in 7838 ms, size: 409 fields circuitName=private-kernel-empty circuitSize=1048576 duration=7837.180525000207 inputSize=16414 proofSize=34308 eventName=circuit-proving numPublicInputs=679
aztec-prover-agent-1   | 2024-08-12T15:35:43.562Z aztec:bb-prover [VERBOSE] BB out - Executing BB with: verify_ultra_honk -p /usr/src/yarn-project/bb/tmp-5UmfCg/proof -k /usr/src/yarn-project/bb/tmp-5UmfCg/vk
aztec-prover-agent-1   | 2024-08-12T15:35:43.603Z aztec:bb-prover [INFO] Successfully verified proof from key in 41.055091000162065 ms
aztec-prover-agent-1   | 2024-08-12T15:35:43.604Z aztec:prover-client:prover-agent [VERBOSE] Processed proving job id=7b2dc62a type=PRIVATE_KERNEL_EMPTY duration=9299.110493000597ms
aztec-node-1           | 2024-08-12T15:35:43.740Z aztec:prover:proving-orchestrator [INFO] Successfully proven block 18!
aztec-prover-1         | 2024-08-12T15:35:44.036Z aztec:archiver [VERBOSE] Retrieved 1 new L2 blocks between L1 blocks 27 and 27.
aztec-prover-1         | 2024-08-12T15:35:44.042Z aztec:archiver [VERBOSE] Processed 1 new L2 blocks up to 18
aztec-prover-1         | 2024-08-12T15:35:44.043Z aztec:archiver [VERBOSE] Synced to L1 block 27
aztec-node-1           | 2024-08-12T15:35:44.136Z aztec:kv-store:lmdb [INFO] Opening LMDB database at temporary location
aztec-node-1           | 2024-08-12T15:35:44.480Z aztec:node [INFO] Simulating tx 0e9a0abd985bdef8461f11179286b44f5101239fe328a1b11f5e3e9554ee8adb
aztec-bot-1            | 2024-08-12T15:35:44.515Z aztec:pxe_service [INFO] Executed local simulation for 0e9a0abd985bdef8461f11179286b44f5101239fe328a1b11f5e3e9554ee8adb
aztec-node-1           | 2024-08-12T15:35:44.773Z aztec:sequencer:publisher [INFO] Published L2 block to L1 rollup contract gasPrice=1159289938 gasUsed=632488 transactionHash=0xa598db4494597a361e1e5122f5ab47559db9cf4aec4778dc8812ed072e9fcf3f calldataGas=6668 calldataSize=740 txCount=0 blockNumber=18 noteEncryptedLogLength=0 noteEncryptedLogCount=0 encryptedLogLength=0 encryptedLogCount=0 unencryptedLogCount=0 unencryptedLogSize=0 eventName=rollup-published-to-l1 blockHash=0x2a8fe0a04fdf0ef084d361ee81659e3554b8a67a47ec61bc6dddf14aee8e5c93
aztec-node-1           | 2024-08-12T15:35:44.773Z aztec:sequencer [INFO] Submitted rollup block 18 with 0 transactions duration=1323.4801650000736ms (Submitter: 0x15d34aaf54267db7d7c367839aaf71a00a2c6a65)
aztec-node-1           | file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32
aztec-node-1           |             throw new Error('Block number mismatch. Expected: ' + expectedL2BlockNumber + ' but got: ' + blockNum + '.');
aztec-node-1           |                   ^
aztec-node-1           | 
aztec-node-1           | Error: Block number mismatch. Expected: 19 but got: 18.
aztec-node-1           |     at processL2BlockProcessedLogs (file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32:19)
aztec-node-1           |     at retrieveBlockMetadataFromRollup (file:///usr/src/yarn-project/archiver/dest/archiver/data_retrieval.js:25:40)
aztec-node-1           |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
aztec-node-1           |     at async Archiver.sync (file:///usr/src/yarn-project/archiver/dest/archiver/archiver.js:134:44)
aztec-node-1           |     at async poll (file:///usr/src/yarn-project/foundation/dest/promise/running-promise.js:22:17)
aztec-node-1           | 
aztec-node-1           | Node.js v18.19.1
aztec-prover-1         | file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32
aztec-prover-1         |             throw new Error('Block number mismatch. Expected: ' + expectedL2BlockNumber + ' but got: ' + blockNum + '.');
aztec-prover-1         |                   ^
aztec-prover-1         | 
aztec-prover-1         | Error: Block number mismatch. Expected: 19 but got: 18.
aztec-prover-1         |     at processL2BlockProcessedLogs (file:///usr/src/yarn-project/archiver/dest/archiver/eth_log_handlers.js:32:19)
aztec-prover-1         |     at retrieveBlockMetadataFromRollup (file:///usr/src/yarn-project/archiver/dest/archiver/data_retrieval.js:25:40)
aztec-prover-1         |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
aztec-prover-1         |     at async Archiver.sync (file:///usr/src/yarn-project/archiver/dest/archiver/archiver.js:134:44)
aztec-prover-1         |     at async poll (file:///usr/src/yarn-project/foundation/dest/promise/running-promise.js:22:17)
aztec-prover-1         | 
aztec-prover-1         | Node.js v18.19.1
aztec-prover-agent-1   | 2024-08-12T15:35:45.337Z aztec:foundation:retry [VERBOSE] JsonRpcClient request provingJobSource_heartbeat to http://aztec-prover failed. Will retry in 1s...
aztec-prover-agent-1   | 2024-08-12T15:35:45.337Z aztec:foundation:retry [ERROR] TypeError: fetch failed
aztec-prover-agent-1   |     at Object.fetch (node:internal/deps/undici/undici:11731:11)
aztec-prover-agent-1   |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
aztec-prover-agent-1   |     at async defaultFetch (file:///usr/src/yarn-project/foundation/dest/json-rpc/client/json_rpc_client.js:29:16)
aztec-prover-agent-1   |     at async retry (file:///usr/src/yarn-project/foundation/dest/retry/index.js:47:20)
aztec-prover-agent-1   |     at async file:///usr/src/yarn-project/foundation/dest/json-rpc/client/json_rpc_client.js:64:16
aztec-prover-agent-1   |     at async request (file:///usr/src/yarn-project/foundation/dest/json-rpc/client/json_rpc_client.js:88:21)
aztec-prover-agent-1   |     at async RunningPromise.fn (file:///usr/src/yarn-project/prover-client/dest/prover-agent/prover-agent.js:43:17)
aztec-prover-agent-1   |     at async poll (file:///usr/src/yarn-project/foundation/dest/promise/running-promise.js:22:17) {
aztec-prover-agent-1   |   cause: [Error]
aztec-prover-agent-1   | }
```

--- a/yarn-project/archiver/src/archiver/archiver.ts
+++ b/yarn-project/archiver/src/archiver/archiver.ts
@@ -145,8 +145,19 @@ export class Archiver implements ArchiveSource {
       await this.sync(blockUntilSynced);
     }
 
-    this.runningPromise = new RunningPromise(() => this.sync(false), this.pollingIntervalMs);
+    this.runningPromise = new RunningPromise(() => this.safeSync(), this.pollingIntervalMs);
     this.runningPromise.start();
+  }
+
+  /**
+   * Syncs and catches exceptions.
+   */
+  private async safeSync() {
+    try {
+      await this.sync(false);
+    } catch (error) {
+      this.log.error('Error syncing archiver', error);
+    }
   }
 
   /**
@@ -166,10 +177,14 @@ export class Archiver implements ArchiveSource {
      *
      * This code does not handle reorgs.
      */
-    const { blocksSynchedTo, messagesSynchedTo } = await this.store.getSynchPoint();
+    const { blockBodiesSynchedTo, blocksSynchedTo, messagesSynchedTo } = await this.store.getSynchPoint();
     const currentL1BlockNumber = await this.publicClient.getBlockNumber();
 
-    if (currentL1BlockNumber <= blocksSynchedTo && currentL1BlockNumber <= messagesSynchedTo) {
+    if (
+      currentL1BlockNumber <= blocksSynchedTo &&
+      currentL1BlockNumber <= messagesSynchedTo &&
+      currentL1BlockNumber <= blockBodiesSynchedTo
+    ) {
       // chain hasn't moved forward
       // or it's been rolled back
       this.log.debug(`Nothing to sync`, { currentL1BlockNumber, blocksSynchedTo, messagesSynchedTo });
@@ -220,16 +235,19 @@ export class Archiver implements ArchiveSource {
     // Read all data from chain and then write to our stores at the end
     const nextExpectedL2BlockNum = BigInt((await this.store.getSynchedL2BlockNumber()) + 1);
 
+    this.log.debug(`Retrieving block bodies from ${blockBodiesSynchedTo + 1n} to ${currentL1BlockNumber}`);
     const retrievedBlockBodies = await retrieveBlockBodiesFromAvailabilityOracle(
       this.publicClient,
       this.availabilityOracleAddress,
       blockUntilSynced,
-      blocksSynchedTo + 1n,
+      blockBodiesSynchedTo + 1n,
       currentL1BlockNumber,
     );
 
-    const blockBodies = retrievedBlockBodies.retrievedData.map(([blockBody]) => blockBody);
-    await this.store.addBlockBodies(blockBodies);
+    this.log.debug(
+      `Retrieved ${retrievedBlockBodies.retrievedData.length} block bodies up to L1 block ${retrievedBlockBodies.lastProcessedL1BlockNumber}`,
+    );
+    await this.store.addBlockBodies(retrievedBlockBodies);
 
     // Now that we have block bodies we will retrieve block metadata and build L2 blocks from the bodies and
     // the metadata
@@ -237,6 +255,7 @@ export class Archiver implements ArchiveSource {
     {
       // @todo @LHerskind Investigate how necessary that nextExpectedL2BlockNum really is.
       //                  Also, I would expect it to break horribly if we have a reorg.
+      this.log.debug(`Retrieving block metadata from ${blocksSynchedTo + 1n} to ${currentL1BlockNumber}`);
       const retrievedBlockMetadata = await retrieveBlockMetadataFromRollup(
         this.publicClient,
         this.rollupAddress,
@@ -278,16 +297,17 @@ export class Archiver implements ArchiveSource {
         } and ${currentL1BlockNumber}.`,
       );
 
-      // Set the `lastProcessedL1BlockNumber` to the smallest of the header and body retrieval
-      const min = (a: bigint, b: bigint) => (a < b ? a : b);
       retrievedBlocks = {
-        lastProcessedL1BlockNumber: min(
-          retrievedBlockMetadata.lastProcessedL1BlockNumber,
-          retrievedBlockBodies.lastProcessedL1BlockNumber,
-        ),
+        lastProcessedL1BlockNumber: retrievedBlockMetadata.lastProcessedL1BlockNumber,
         retrievedData: blocks,
       };
     }
+
+    this.log.debug(
+      `Processing retrieved blocks ${retrievedBlocks.retrievedData
+        .map(b => b.number)
+        .join(',')} with last processed L1 block ${retrievedBlocks.lastProcessedL1BlockNumber}`,
+    );
 
     await Promise.all(
       retrievedBlocks.retrievedData.map(block => {

--- a/yarn-project/archiver/src/archiver/archiver_store.ts
+++ b/yarn-project/archiver/src/archiver/archiver_store.ts
@@ -30,8 +30,10 @@ import { type DataRetrieval } from './data_retrieval.js';
  * Represents the latest L1 block processed by the archiver for various objects in L2.
  */
 export type ArchiverL1SynchPoint = {
-  /** Number of the last L1 block that added a new L2 block.  */
+  /** Number of the last L1 block that added a new L2 block metadata.  */
   blocksSynchedTo: bigint;
+  /** Number of the last L1 block that added a new L2 block body.  */
+  blockBodiesSynchedTo: bigint;
   /** Number of the last L1 block that added L1 -> L2 messages from the Inbox. */
   messagesSynchedTo: bigint;
 };
@@ -53,7 +55,7 @@ export interface ArchiverDataStore {
    * @param blockBodies - The L2 block bodies to be added to the store.
    * @returns True if the operation is successful.
    */
-  addBlockBodies(blockBodies: Body[]): Promise<boolean>;
+  addBlockBodies(blockBodies: DataRetrieval<Body>): Promise<boolean>;
 
   /**
    * Gets block bodies that have the same txsEffectsHashes as we supply.

--- a/yarn-project/archiver/src/archiver/archiver_store_test_suite.ts
+++ b/yarn-project/archiver/src/archiver/archiver_store_test_suite.ts
@@ -1,4 +1,4 @@
-import { InboxLeaf, L2Block, LogId, LogType, TxHash } from '@aztec/circuit-types';
+import { type Body, InboxLeaf, L2Block, LogId, LogType, TxHash } from '@aztec/circuit-types';
 import '@aztec/circuit-types/jest';
 import { AztecAddress, Fr, INITIAL_L2_BLOCK_NUM, L1_TO_L2_MSG_SUBTREE_HEIGHT } from '@aztec/circuits.js';
 import {
@@ -14,7 +14,7 @@ import {
   SerializableContractInstance,
 } from '@aztec/types/contracts';
 
-import { type ArchiverDataStore } from './archiver_store.js';
+import { type ArchiverDataStore, type ArchiverL1SynchPoint } from './archiver_store.js';
 import { type DataRetrieval } from './data_retrieval.js';
 
 /**
@@ -25,6 +25,7 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
   describe(testName, () => {
     let store: ArchiverDataStore;
     let blocks: DataRetrieval<L2Block>;
+    let blockBodies: DataRetrieval<Body>;
     const blockTests: [number, number, () => L2Block[]][] = [
       [1, 1, () => blocks.retrievedData.slice(0, 1)],
       [10, 1, () => blocks.retrievedData.slice(9, 10)],
@@ -39,11 +40,15 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
         lastProcessedL1BlockNumber: 5n,
         retrievedData: Array.from({ length: 10 }).map((_, i) => L2Block.random(i + 1)),
       };
+      blockBodies = {
+        retrievedData: blocks.retrievedData.map(block => block.body),
+        lastProcessedL1BlockNumber: 4n,
+      };
     });
 
     describe('addBlocks', () => {
       it('returns success when adding block bodies', async () => {
-        await expect(store.addBlockBodies(blocks.retrievedData.map(block => block.body))).resolves.toBe(true);
+        await expect(store.addBlockBodies(blockBodies)).resolves.toBe(true);
       });
 
       it('returns success when adding blocks', async () => {
@@ -59,7 +64,7 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
     describe('getBlocks', () => {
       beforeEach(async () => {
         await store.addBlocks(blocks);
-        await store.addBlockBodies(blocks.retrievedData.map(block => block.body));
+        await store.addBlockBodies(blockBodies);
       });
 
       it.each(blockTests)('retrieves previously stored blocks', async (start, limit, getExpectedBlocks) => {
@@ -95,7 +100,8 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
         await expect(store.getSynchPoint()).resolves.toEqual({
           blocksSynchedTo: 0n,
           messagesSynchedTo: 0n,
-        });
+          blockBodiesSynchedTo: 0n,
+        } satisfies ArchiverL1SynchPoint);
       });
 
       it('returns the L1 block number in which the most recent L2 block was published', async () => {
@@ -103,7 +109,17 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
         await expect(store.getSynchPoint()).resolves.toEqual({
           blocksSynchedTo: blocks.lastProcessedL1BlockNumber,
           messagesSynchedTo: 0n,
-        });
+          blockBodiesSynchedTo: 0n,
+        } satisfies ArchiverL1SynchPoint);
+      });
+
+      it('returns the L1 block number in which the most recent L2 block body was published', async () => {
+        await store.addBlockBodies(blockBodies);
+        await expect(store.getSynchPoint()).resolves.toEqual({
+          blocksSynchedTo: 0n,
+          messagesSynchedTo: 0n,
+          blockBodiesSynchedTo: blockBodies.lastProcessedL1BlockNumber,
+        } satisfies ArchiverL1SynchPoint);
       });
 
       it('returns the L1 block number that most recently added messages from inbox', async () => {
@@ -114,7 +130,8 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
         await expect(store.getSynchPoint()).resolves.toEqual({
           blocksSynchedTo: 0n,
           messagesSynchedTo: 1n,
-        });
+          blockBodiesSynchedTo: 0n,
+        } satisfies ArchiverL1SynchPoint);
       });
     });
 
@@ -179,7 +196,7 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
           ),
         );
         await store.addBlocks(blocks);
-        await store.addBlockBodies(blocks.retrievedData.map(block => block.body));
+        await store.addBlockBodies(blockBodies);
       });
 
       it.each([
@@ -335,7 +352,7 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
         };
 
         await store.addBlocks(blocks);
-        await store.addBlockBodies(blocks.retrievedData.map(block => block.body));
+        await store.addBlockBodies(blockBodies);
 
         await Promise.all(
           blocks.retrievedData.map(block =>

--- a/yarn-project/archiver/src/archiver/data_retrieval.ts
+++ b/yarn-project/archiver/src/archiver/data_retrieval.ts
@@ -97,7 +97,6 @@ export async function retrieveBlockBodiesFromAvailabilityOracle(
   searchEndBlock: bigint,
 ): Promise<DataRetrieval<Body>> {
   const retrievedBlockBodies: Body[] = [];
-  let lastProcessedL1BlockNumber = searchStartBlock;
 
   do {
     if (searchStartBlock > searchEndBlock) {
@@ -115,11 +114,10 @@ export async function retrieveBlockBodiesFromAvailabilityOracle(
 
     const newBlockBodies = await processTxsPublishedLogs(publicClient, l2TxsPublishedLogs);
     retrievedBlockBodies.push(...newBlockBodies.map(([body]) => body));
-    lastProcessedL1BlockNumber = l2TxsPublishedLogs[l2TxsPublishedLogs.length - 1].blockNumber!;
-    searchStartBlock = lastProcessedL1BlockNumber + 1n;
+    searchStartBlock = l2TxsPublishedLogs[l2TxsPublishedLogs.length - 1].blockNumber + 1n;
   } while (blockUntilSynced && searchStartBlock <= searchEndBlock);
 
-  return { lastProcessedL1BlockNumber, retrievedData: retrievedBlockBodies };
+  return { lastProcessedL1BlockNumber: searchStartBlock - 1n, retrievedData: retrievedBlockBodies };
 }
 
 /**

--- a/yarn-project/archiver/src/archiver/data_retrieval.ts
+++ b/yarn-project/archiver/src/archiver/data_retrieval.ts
@@ -1,6 +1,7 @@
 import { type Body, type InboxLeaf } from '@aztec/circuit-types';
 import { type AppendOnlyTreeSnapshot, Fr, type Header } from '@aztec/circuits.js';
 import { type EthAddress } from '@aztec/foundation/eth-address';
+import { type DebugLogger, createDebugLogger } from '@aztec/foundation/log';
 import { RollupAbi } from '@aztec/l1-artifacts';
 
 import { type PublicClient, getAbiItem } from 'viem';
@@ -45,6 +46,7 @@ export async function retrieveBlockMetadataFromRollup(
   searchStartBlock: bigint,
   searchEndBlock: bigint,
   expectedNextL2BlockNum: bigint,
+  logger: DebugLogger = createDebugLogger('aztec:archiver'),
 ): Promise<DataRetrieval<[Header, AppendOnlyTreeSnapshot]>> {
   const retrievedBlockMetadata: [Header, AppendOnlyTreeSnapshot][] = [];
   do {
@@ -61,13 +63,18 @@ export async function retrieveBlockMetadataFromRollup(
       break;
     }
 
+    const lastLog = l2BlockProcessedLogs[l2BlockProcessedLogs.length - 1];
+    logger.debug(
+      `Got L2 block processed logs for ${l2BlockProcessedLogs[0].blockNumber}-${lastLog.blockNumber} between ${searchStartBlock}-${searchEndBlock} L1 blocks`,
+    );
+
     const newBlockMetadata = await processL2BlockProcessedLogs(
       publicClient,
       expectedNextL2BlockNum,
       l2BlockProcessedLogs,
     );
     retrievedBlockMetadata.push(...newBlockMetadata);
-    searchStartBlock = l2BlockProcessedLogs[l2BlockProcessedLogs.length - 1].blockNumber! + 1n;
+    searchStartBlock = lastLog.blockNumber! + 1n;
     expectedNextL2BlockNum += BigInt(newBlockMetadata.length);
   } while (blockUntilSynced && searchStartBlock <= searchEndBlock);
   return { lastProcessedL1BlockNumber: searchStartBlock - 1n, retrievedData: retrievedBlockMetadata };
@@ -80,7 +87,7 @@ export async function retrieveBlockMetadataFromRollup(
  * @param blockUntilSynced - If true, blocks until the archiver has fully synced.
  * @param searchStartBlock - The block number to use for starting the search.
  * @param searchEndBlock - The highest block number that we should search up to.
- * @returns A array of tuples of L2 block bodies and their associated hash as well as the next eth block to search from
+ * @returns A array of L2 block bodies as well as the next eth block to search from
  */
 export async function retrieveBlockBodiesFromAvailabilityOracle(
   publicClient: PublicClient,
@@ -88,8 +95,9 @@ export async function retrieveBlockBodiesFromAvailabilityOracle(
   blockUntilSynced: boolean,
   searchStartBlock: bigint,
   searchEndBlock: bigint,
-): Promise<DataRetrieval<[Body, Buffer]>> {
-  const retrievedBlockBodies: [Body, Buffer][] = [];
+): Promise<DataRetrieval<Body>> {
+  const retrievedBlockBodies: Body[] = [];
+  let lastProcessedL1BlockNumber = searchStartBlock;
 
   do {
     if (searchStartBlock > searchEndBlock) {
@@ -106,10 +114,12 @@ export async function retrieveBlockBodiesFromAvailabilityOracle(
     }
 
     const newBlockBodies = await processTxsPublishedLogs(publicClient, l2TxsPublishedLogs);
-    retrievedBlockBodies.push(...newBlockBodies);
-    searchStartBlock = l2TxsPublishedLogs[l2TxsPublishedLogs.length - 1].blockNumber! + 1n;
+    retrievedBlockBodies.push(...newBlockBodies.map(([body]) => body));
+    lastProcessedL1BlockNumber = l2TxsPublishedLogs[l2TxsPublishedLogs.length - 1].blockNumber!;
+    searchStartBlock = lastProcessedL1BlockNumber + 1n;
   } while (blockUntilSynced && searchStartBlock <= searchEndBlock);
-  return { lastProcessedL1BlockNumber: searchStartBlock - 1n, retrievedData: retrievedBlockBodies };
+
+  return { lastProcessedL1BlockNumber, retrievedData: retrievedBlockBodies };
 }
 
 /**

--- a/yarn-project/archiver/src/archiver/kv_archiver_store/block_body_store.test.ts
+++ b/yarn-project/archiver/src/archiver/kv_archiver_store/block_body_store.test.ts
@@ -13,12 +13,14 @@ describe('Block Body Store', () => {
   it('Should add and return block bodies', async () => {
     const body = Body.random(1);
 
-    await archiverStore.addBlockBodies([body]);
+    await archiverStore.addBlockBodies({ retrievedData: [body], lastProcessedL1BlockNumber: 5n });
 
     const txsEffectsHash = body.getTxsEffectsHash();
 
     const [returnedBody] = await archiverStore.getBlockBodies([txsEffectsHash]);
-
     expect(body).toStrictEqual(returnedBody);
+
+    const { blockBodiesSynchedTo } = await archiverStore.getSynchPoint();
+    expect(blockBodiesSynchedTo).toEqual(5n);
   });
 });

--- a/yarn-project/archiver/src/archiver/kv_archiver_store/block_body_store.ts
+++ b/yarn-project/archiver/src/archiver/kv_archiver_store/block_body_store.ts
@@ -1,13 +1,19 @@
 import { Body } from '@aztec/circuit-types';
 import { createDebugLogger } from '@aztec/foundation/log';
-import { type AztecKVStore, type AztecMap } from '@aztec/kv-store';
+import { type AztecKVStore, type AztecMap, type AztecSingleton } from '@aztec/kv-store';
+
+import { type DataRetrieval } from '../data_retrieval.js';
 
 export class BlockBodyStore {
   /** Map block body hash to block body */
   #blockBodies: AztecMap<string, Buffer>;
 
+  /** Stores L1 block number in which the last processed L2 block body was included */
+  #lastSynchedL1Block: AztecSingleton<bigint>;
+
   constructor(private db: AztecKVStore, private log = createDebugLogger('aztec:archiver:block_body_store')) {
     this.#blockBodies = db.openMap('archiver_block_bodies');
+    this.#lastSynchedL1Block = db.openSingleton('archiver_block_bodies_last_synched_l1_block');
   }
 
   /**
@@ -15,12 +21,12 @@ export class BlockBodyStore {
    * @param blockBodies - The L2 block bodies to be added to the store.
    * @returns True if the operation is successful.
    */
-  addBlockBodies(blockBodies: Body[]): Promise<boolean> {
+  addBlockBodies(blockBodies: DataRetrieval<Body>): Promise<boolean> {
     return this.db.transaction(() => {
-      for (const body of blockBodies) {
+      for (const body of blockBodies.retrievedData) {
         void this.#blockBodies.set(body.getTxsEffectsHash().toString('hex'), body.toBuffer());
       }
-
+      void this.#lastSynchedL1Block.set(blockBodies.lastProcessedL1BlockNumber);
       return true;
     });
   }
@@ -56,5 +62,13 @@ export class BlockBodyStore {
     const blockBody = this.#blockBodies.get(txsEffectsHash.toString('hex'));
 
     return blockBody && Body.fromBuffer(blockBody);
+  }
+
+  /**
+   * Gets the last L1 block number in which a L2 block body was included
+   * @returns The L1 block number
+   */
+  getSynchedL1BlockNumber(): bigint {
+    return this.#lastSynchedL1Block.get() ?? 0n;
   }
 }

--- a/yarn-project/archiver/src/archiver/kv_archiver_store/kv_archiver_store.ts
+++ b/yarn-project/archiver/src/archiver/kv_archiver_store/kv_archiver_store.ts
@@ -101,7 +101,7 @@ export class KVArchiverDataStore implements ArchiverDataStore {
    * @param blockBodies - The L2 block bodies to be added to the store.
    * @returns True if the operation is successful.
    */
-  addBlockBodies(blockBodies: Body[]): Promise<boolean> {
+  addBlockBodies(blockBodies: DataRetrieval<Body>): Promise<boolean> {
     return this.#blockBodyStore.addBlockBodies(blockBodies);
   }
 
@@ -260,6 +260,7 @@ export class KVArchiverDataStore implements ArchiverDataStore {
   getSynchPoint(): Promise<ArchiverL1SynchPoint> {
     return Promise.resolve({
       blocksSynchedTo: this.#blockStore.getSynchedL1BlockNumber(),
+      blockBodiesSynchedTo: this.#blockBodyStore.getSynchedL1BlockNumber(),
       messagesSynchedTo: this.#messageStore.getSynchedL1BlockNumber(),
     });
   }

--- a/yarn-project/archiver/src/archiver/memory_archiver_store/memory_archiver_store.ts
+++ b/yarn-project/archiver/src/archiver/memory_archiver_store/memory_archiver_store.ts
@@ -83,6 +83,7 @@ export class MemoryArchiverStore implements ArchiverDataStore {
   private contractInstances: Map<string, ContractInstanceWithAddress> = new Map();
 
   private lastL1BlockNewBlocks: bigint = 0n;
+  private lastL1BlockNewBlockBodies: bigint = 0n;
   private lastL1BlockNewMessages: bigint = 0n;
   private lastProvenL2BlockNumber: number = 0;
 
@@ -163,11 +164,11 @@ export class MemoryArchiverStore implements ArchiverDataStore {
    * @param blockBodies - The L2 block bodies to be added to the store.
    * @returns True if the operation is successful.
    */
-  addBlockBodies(blockBodies: Body[]): Promise<boolean> {
-    for (const body of blockBodies) {
+  addBlockBodies(blockBodies: DataRetrieval<Body>): Promise<boolean> {
+    for (const body of blockBodies.retrievedData) {
       void this.l2BlockBodies.set(body.getTxsEffectsHash().toString('hex'), body);
     }
-
+    this.lastL1BlockNewBlockBodies = blockBodies.lastProcessedL1BlockNumber;
     return Promise.resolve(true);
   }
 
@@ -443,6 +444,7 @@ export class MemoryArchiverStore implements ArchiverDataStore {
     return Promise.resolve({
       blocksSynchedTo: this.lastL1BlockNewBlocks,
       messagesSynchedTo: this.lastL1BlockNewMessages,
+      blockBodiesSynchedTo: this.lastL1BlockNewBlockBodies,
     });
   }
 

--- a/yarn-project/circuit-types/src/l2_block.ts
+++ b/yarn-project/circuit-types/src/l2_block.ts
@@ -233,6 +233,7 @@ export class L2Block {
     return {
       txCount: this.body.txEffects.length,
       blockNumber: this.number,
+      blockTimestamp: this.header.globalVariables.timestamp.toNumber(),
       ...logsStats,
     };
   }

--- a/yarn-project/end-to-end/src/e2e_block_building.test.ts
+++ b/yarn-project/end-to-end/src/e2e_block_building.test.ts
@@ -11,6 +11,7 @@ import {
   type PXE,
   type Wallet,
   deriveKeys,
+  retryUntil,
 } from '@aztec/aztec.js';
 import { times } from '@aztec/foundation/collection';
 import { poseidon2HashWithSeparator } from '@aztec/foundation/crypto';
@@ -314,6 +315,17 @@ describe('e2e_block_building', () => {
       if (teardown) {
         await teardown();
       }
+    });
+
+    // Regression for https://github.com/AztecProtocol/aztec-packages/issues/7918
+    it('publishes two blocks with only padding txs', async () => {
+      ({ teardown, pxe, logger, aztecNode } = await setup(0, {
+        minTxsPerBlock: 0,
+        sequencerSkipSubmitProofs: true,
+        skipProtocolContracts: true,
+      }));
+
+      await retryUntil(async () => (await aztecNode.getBlockNumber()) >= 3, 'wait-block', 10, 1);
     });
 
     // Regression for https://github.com/AztecProtocol/aztec-packages/issues/7537

--- a/yarn-project/sequencer-client/src/sequencer/sequencer.ts
+++ b/yarn-project/sequencer-client/src/sequencer/sequencer.ts
@@ -201,11 +201,14 @@ export class Sequencer {
       const lastBlockTime = historicalHeader?.globalVariables.timestamp.toNumber() || 0;
       const currentTime = Math.floor(Date.now() / 1000);
       const elapsedSinceLastBlock = currentTime - lastBlockTime;
+      this.log.debug(
+        `Last block mined at ${lastBlockTime} current time is ${currentTime} (elapsed ${elapsedSinceLastBlock})`,
+      );
 
       // Do not go forward with new block if not enough time has passed since last block
       if (this.minSecondsBetweenBlocks > 0 && elapsedSinceLastBlock < this.minSecondsBetweenBlocks) {
         this.log.debug(
-          `Not creating block because not enough time has passed since last block (last block at ${lastBlockTime} current time ${currentTime})`,
+          `Not creating block because not enough time ${this.minSecondsBetweenBlocks} has passed since last block`,
         );
         return;
       }
@@ -334,7 +337,7 @@ export class Sequencer {
     // less txs than the minimum. But that'd cause the entire block to be aborted and retried. Instead, we should
     // go back to the p2p pool and load more txs until we hit our minTxsPerBLock target. Only if there are no txs
     // we should bail.
-    if (processedTxs.length === 0 && !this.skipMinTxsPerBlockCheck(elapsedSinceLastBlock)) {
+    if (processedTxs.length === 0 && !this.skipMinTxsPerBlockCheck(elapsedSinceLastBlock) && this.minTxsPerBLock > 0) {
       this.log.verbose('No txs processed correctly to build block. Exiting');
       prover.cancelBlock();
       return;

